packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -143,10 +143,20 @@ const App = (props) => {
         l10nConfig: site.l10n
     })
 
+    // Workaround for infinite loop if default site locale translation file is missing
+    const isDefaultTranslation = new RegExp(
+        `/static/translations/compiled/${site.l10n.defaultLocale}`
+    ).test(location?.pathname)
```

Comment:
This is the main workaround. If `isDefaultTranslation` is true, then we don't call fetchTranslations and go straight to returning an empty object.

Reply:
Also, this uses `site.l10n.defaultLocale` because this can be different from DEFAULT_LOCALE

In our codebase, `site.l10n.defaultLocale` is en-GB since we develop on RefArchGlobal while DEFAULT_LOCALE (set in app/constants) is en-US

We may want to align these two so they are the same locale?

Reply:
Presumably we'd want the out-of-the-box DEFAULT_LOCALE to be the most common one used by our customers. Would that be en-US or something different?

Also, DEFAULT_LOCALE seems to be used exclusively as a last-resort fallback if locale isn't detected any other way. Maybe having it be different from our `site.l10n.defaultLocale` will help us avoid accidentally using DEFAULT_LOCALE incorrectly? 🤷

Reply:
@vcua-mobify default-locale concept means differently, depending on whether you're talking about BM or `react-intl` library:
- Like you said, in BM, the default locale can vary depending on the active site.
- While react-intl does not know about those locales in BM. It cares only about the locale of the hardcoded strings in your app, which we defined it as en-US. 

So in this case, we should be using DEFAULT_LOCALE instead.